crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -462,8 +462,7 @@ where
     ) -> bool {
         // On Optimism, the proposers are allowed to reorg their own chain at will.
         #[cfg(feature = "optimism")]
-        if reth_chainspec::EthChainSpec::chain(self.blockchain.chain_spec().as_ref()).is_optimism()
-        {
+        if reth_chainspec::EthChainSpec::is_optimism(&self.blockchain.chain_spec()) {
```

Comment:
we should open an issue to move away from this type of behaviour, since it requires every rollup to add a new `is_my_rollup` trait method to `EthChainSpec`, and instead solve it at sdk level for engine